### PR TITLE
Update mason: Compile from sources (instead of repackaging the binary)

### PR DESCRIPTION
**Pull request by marcelm**
The pre-compiled binary for mason_simulator (at least the Linux one) has seemingly been compiled with assertions enabled, which makes it very slow.

Also, often one of these assertions is triggered and crashes the program.

See https://github.com/seqan/seqan/issues/2520

Describe your pull request here

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/mason/build.sh [hunk 1]
```diff
@@ -1,14 +1,5 @@
...
-mason_methylation \
-mason_simulator \
-mason_splicing \
-mason_variator \
-"
-mkdir -p $PREFIX/bin
-
-for i in $binaries; do cp bin/$i $PREFIX/bin/$i && chmod a+x $PREFIX/bin/$i; done
+cmake -S . -B build -DSEQAN_BUILD_SYSTEM=APP:mason2 -DCMAKE_INSTALL_PREFIX=${PREFIX}
+cmake --build build/
```
Comment (martin-g) [inline recipes/mason/build.sh:4]:
```suggestion
cmake --build build/ -j ${CPU_COUNT}
```

File: recipes/mason/meta.yaml [hunk 2]
```diff
@@ -5,13 +5,23 @@ package:
   version: {{ version }}
 
 source:
-  - url:  http://packages.seqan.de/mason2/mason2-{{ version }}-Mac-x86_64_sse4.zip      # [osx]
-    sha256: d478b07effe6dd2056ead6cd816fa2b196ba1e2ffb9d6d8d98e24ad968834ee9            # [osx]
-  - url: http://packages.seqan.de/mason2/mason2-{{ version }}-Linux-x86_64_sse4.tar.xz  # [linux]
-    sha256: 3879cdd71c39947d9fc731e820c9b926912e7688d070a8431c75681848eee688            # [linux]
+  - url:  http://packages.seqan.de/seqan-src/seqan-src-2.4.0.tar.gz
```
Comment (martin-g) [inline recipes/mason/meta.yaml:8]: Please update the recipe version too

Comment (marcelm) [reply recipes/mason/meta.yaml:10]: I have added a comment to the recipe to point this out.

**Comment by marcelm** [comment]
@martin-g Thanks for the review! If you have time: I seem to have fixed the macOS compilation failure, the build passes now.

**Review by martin-g — APPROVED** [review]

**Pull request merged**